tests/test-p4-master-site.php

```diff
@@ -16,42 +16,45 @@
-			],
-			[
-				[ 'http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE.jpg 1200w, http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE-300x210.jpg 300w, http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE-1024x715.jpg 1024w, http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE-768x536.jpg 768w, http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE-487x340.jpg 487w', '', 'optionA=valA' ],
-				'/cdn-cgi/image/format=auto,optionA=valA,fit=contain,width=1200/http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE.jpg 1200w, /cdn-cgi/image/format=auto,optionA=valA,fit=contain,width=300/http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE-300x210.jpg 300w, /cdn-cgi/image/format=auto,optionA=valA,fit=contain,width=1024/http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE-1024x715.jpg 1024w, /cdn-cgi/image/format=auto,optionA=valA,fit=contain,width=768/http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE-768x536.jpg 768w, /cdn-cgi/image/format=auto,optionA=valA,fit=contain,width=487/http://www.planet4.test/wp-content/uploads/2020/06/GP1STWPE-487x340.jpg 487w',
+			'basic test'          => [ '<img src="img-url" />', '<img class="lazyload" data-src="img-url" />' ],
+			'all attrs'           => [
+				'<img src="img-url" srcset="img-srcset" sizes="img-srcsizes" />',
+				'<img class="lazyload" data-src="img-url" data-srcset="img-srcset" data-sizes="img-srcsizes" />',
 			],
+			'll present 1'        => [ '<img src="img-url" class="lazyload" />', null ],
+			'll present 2'        => [ '<img src="img-url" class="foo lazyload" />', null ],
+			'll present 3'        => [ '<img src="img-url" class="lazyload bar" />', null ],
+			'll present 4'        => [ '<img src="img-url" class="foo lazyload bar" />', null ],
+			'data-src present'    => [ '<img src="img-url" data-src="src" />', null ],
+			'data-srcset present' => [ '<img src="img-url" data-srcset="srcset" />', null ],
+			'data-sizes'          => [ '<img src="img-url" data-sizes="sizes" />', null ],
 		];
 	}
 
 	/**
-	 * @dataProvider cloudflareImageProvider
+	 * @dataProvider contentImageProvider
 	 *
-	 * @param array  $params   Parameters for img_to_cloudflare().
-	 * @param string $expected Resulting URL expected.
+	 * If $expected is null, it will be replaced by $content,
+	 * to check if nothing has changed
+	 *
+	 * @param string  $content Html content given.
+	 * @param ?string $expected Html content expected.
 	 */
-	public function testImgToCloudflare( array $params, string $expected ): void {
+	public function testMakeContentImageLazyLoad( string $content, ?string $expected ): void {
 		/** @var MasterSite $p4 */
 		$p4 = $this->getMockBuilder( MasterSite::class )
-		->disableOriginalConstructor()
-		->setMethodsExcept( [ 'img_to_cloudflare' ] )
-		->getMock();
+			->disableOriginalConstructor()
+			->setMethodsExcept( [ 'make_content_images_lazyload' ] )
```

Comment:
Not sure the revert worked properly here, you should drop all tests, this function was removed in https://github.com/greenpeace/planet4-master-theme/pull/1176 (by you :grin:  https://github.com/greenpeace/planet4-master-theme/commit/1959090029992a5b98d47e001899ab9234dc731a#diff-d5a71432d6e94abc1260a448df306d8dfc76d15fa4047646a4d133c38cc131d7L1127 )

Reply:
Oh yes. I reverted the two commits, then squashed but something went wrong when resolving the conflicts.

Reply:
Commit amended